.github/workflows/build.yml

```diff
@@ -8,7 +8,7 @@ on:
       - "**.md"
 
 env:
-  dotnetVersion: "6.0.x"
+  DOTNET_VERSION: "6.0"
```

Comment:
We cannot remove the `.x` here. This is functionally required to roll forward to the latest minor release of dotnet. See [here](https://github.com/actions/setup-dotnet/blob/c20f59e04ac89e07b0b6d08da3c8694acdfaaf7a/src/installer.ts#L273).

Also since you're passing this version into `build-args` below, I do not thing the rename is necessary. To reduce churn in the file, please use the original name.

I think to get the effect you were aiming for, we should do `dotnetVersion: "6.0"` and in the `setup-dotnet` steps do something like `dotnet-version: ${{ env.dotnetVersion }}.x`. I think that'd get us the best of both worlds. What do you think?

Reply:
> We cannot remove the .x [here](https://github.com/actions/setup-dotnet/blob/c20f59e04ac89e07b0b6d08da3c8694acdfaaf7a/src/installer.ts#L273). This is functionally required to roll forward to the latest minor release of dotnet. See here.

The `.x` is not necessary. I checked the source code and the patch number will be completely ignored, as you can see here, tested with RunJS:

![image](https://user-images.githubusercontent.com/10259118/163006897-cbdeed07-10f3-4728-9c4a-3b0dd926ee1f.png)

The piece of code you've linked only handles versions like `6.x` or `6.*`, as far as I know.

> Also, since you're passing this version into build-args below, I do not think the rename is necessary. To reduce churn in the file, please use the original name.

I've renamed the variable because env vars are usually uppercase, but I will revert it.

> I think to get the effect you were aiming for, we should do dotnetVersion: "6.0" and in the setup-dotnet steps do something like dotnet-version: ${{ env.dotnetVersion }}.x. I think that'd get us the best of both worlds. What do you think?

As explained above, I do not think this will be necessary. But if you prefer to have more clarity or something, I can still add it if you want.

Reply:
Okay so this is now irrelevant because I've moved the Docker build steps to it's own workflow.